bluepyopt/ephys/models.py

```diff
@@ -96,10 +105,20 @@ def __init__(
 
         self.param_values = None
         self.gid = gid
-        self.seclist_names = \
-            ['all', 'somatic', 'basal', 'apical', 'axonal', 'myelinated']
-        self.secarray_names = \
-            ['soma', 'dend', 'apic', 'axon', 'myelin']
+
+        if seclist_names is None:
+            self.seclist_names = [
+                'all', 'somatic', 'basal', 'apical', 'axonal', 'myelinated'
+            ]
+        else:
+            self.seclist_names = seclist_names
+
+        if secarray_names is None:
+            self.secarray_names = [
+                'soma', 'dend', 'apic', 'axon', 'myelin'
+            ]
+        else:
+            self.secarray_names = secarray_names
 
```

Comment:
what about a check that at least soma/dend/axon exists? Can you run something without any of these?

Reply:
Good question, the other option is to only pass the custom names and to append them to the minimal list.